ssl/statem/statem.c

```diff
@@ -951,3 +951,12 @@ int ossl_statem_export_allowed(SSL *s)
     return s->s3->previous_server_finished_len != 0
            && s->statem.hand_state != TLS_ST_SW_FINISHED;
 }
+
+/*
+ * This function returns 1 if early TLS exporter is ready to export
+ * keying material, or 0 if otherwise.
+ */
+int ossl_statem_export_early_allowed(SSL *s)
+{
+    return s->statem.hand_state == TLS_ST_EARLY_DATA;
```

Comment:
Would we ever want to be able to export based on the early secret "after" this point?

Reply:
Hmm, I think spec does not prohibit it.
I used hand_state because it is very convenient that it (I think) ensures that early master secret is generated because there is another code path which does not generate it.
Is there any flag or state variable which indicates that early master secret is generated, so that we can use it after this point?

Reply:
On the server side we set `s->ext.early_data` to `SSL_EARLY_DATA_ACCEPTED` immediately before generating the early exporter secret. On the client side we initially set `s->ext.early_data` to `SSL_EARLY_DATA_REJECTED` and generate the ClientHello and early exporter secret. We later update `s->ext.early_data` to `SSL_EARLY_DATA_ACCEPTED` when the server accepts it. So I think something like this should do it:

````
/*
 * The early exporter secret is only present on the server if we have accepted early_data.
 * It is present on the client as long as we have sent early_data.
 */
return s->ext.early_data == SSL_EARLY_DATA_ACCEPTED || (!s->server && s->ext.early_data != SSL_EARLY_DATA_NOT_SENT);
````

Reply:
Thank you.  Fixed.